charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -240,13 +240,13 @@ spec:
           - name: DD_APM_INSTRUMENTATION_ENABLED
             value: "true"
           {{- end }}
-          {{- if ne (include "apmInstrumentation.enabledNamespaces" .) "" }}
+          {{- if .Values.datadog.apm.instrumentation.enabledNamespaces }}
           - name: DD_APM_INSTRUMENTATION_ENABLED_NAMESPACES
-            value: {{ include "apmInstrumentation.enabledNamespaces" . }}
+            value: {{ .Values.datadog.apm.instrumentation.enabledNamespaces | toJson | quote }}
```

Comment:
I was thinking that namespace list separated by a space that the supported format. So why do we convert it to json now?

Reply:
We are reading these env variables in datadog-agent using [config.Datadog.GetStringSlice(...)](https://github.com/DataDog/datadog-agent/blob/main/pkg/clusteragent/admission/mutate/auto_instrumentation.go#L182). GetStringSlice is implemented to [json unmarshal a string to the slice](https://github.com/DataDog/cast/blob/1ee8c8bd14a3d768a7ff681617ed56bc6c204940/caste.go#L1133).
 
Without converting to json, env variables set on DCA would be:
```
    - name: DD_APM_INSTRUMENTATION_ENABLED_NAMESPACES
      value: '[rc-tests apps]'
    - name: DD_APM_INSTRUMENTATION_LIB_VERSIONS
      value: map[java:v1.22.0 python:v1.20.6]
```
and won't be unmarshalled into the slice.